src/torchmetrics/utilities/plot.py

```diff
@@ -0,0 +1,264 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from itertools import product
+from math import ceil, floor, sqrt
+from typing import List, Optional, Sequence, Tuple, Union
+
+import numpy as np
+import torch
+from torch import Tensor
+
+from torchmetrics.utilities.compute import _auc_compute_without_check as auc_calc
+from torchmetrics.utilities.imports import _MATPLOTLIB_AVAILABLE
+
+if _MATPLOTLIB_AVAILABLE:
+    import matplotlib
+    import matplotlib.pyplot as plt
+
+    _PLOT_OUT_TYPE = Tuple[plt.Figure, matplotlib.axes.Axes]
+else:
+    plt = None
+    _PLOT_OUT_TYPE = None
+
+
+def _error_on_missing_matplotlib() -> None:
+    """Raise error if matplotlib is not installed."""
+    if not _MATPLOTLIB_AVAILABLE:
+        raise ValueError(
+            "Plot function expects `matplotlib` to be installed. Please install with `pip install matplotlib`"
+        )
+
+
+def plot_single_or_multi_val(
+    val: Optional[Union[Tensor, Sequence[Tensor]]],
+    higher_is_better: Optional[bool] = None,
+    lower_bound: Optional[float] = None,
+    upper_bound: Optional[float] = None,
+    legend_name: Optional[str] = None,
+    name: Optional[str] = None,
+    fig_ax: Optional[_PLOT_OUT_TYPE] = None,
```

Comment:
we only need the axes here and can retrieve the figure with `ax.get_figure()` (if we need it at all). This makes it easier to pass things over as plotting should ideally only be done on the axes